google/cloud/bigquery/table.py

```diff
@@ -2942,6 +2953,123 @@ def __repr__(self):
         return "TimePartitioning({})".format(",".join(key_vals))
 
 
+class PrimaryKey:
+    """Represents the primary key constraint on a table's columns.
+
+    Args:
+        columns: The columns that are composed of the primary key constraint.
+    """
+
+    def __init__(self, columns: List[str]):
+        self.columns = columns
+
+    def __eq__(self, other):
+        if not isinstance(other, PrimaryKey):
+            raise NotImplementedError
```

Comment:
My understanding from the Python documentation is that:

> [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError): Is raised when an operation or function is applied to an object of inappropriate type. The associated value is a string giving details about the type mismatch.

> This exception may be raised by user code to indicate that an attempted operation on an object is not supported, and is not meant to be. If an object is meant to support a given operation but has not yet provided an implementation, [NotImplementedError](https://docs.python.org/3/library/exceptions.html#NotImplementedError) is the proper exception to raise.

> Passing arguments of the wrong type (e.g. passing a list when an int is expected) should result in a [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError), but passing arguments with the wrong value (e.g. a number outside expected boundaries) should result in a [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError).

```suggestion
            raise TypeError("The value provided is not a BigQuery PrimaryKey.")
```

Reply:
@chalmerlowe oh, right. Thank you. I missed that. It should be `return` but not `raise`.
I think we should not `raise` anything here to not cause errors and to be consistent with the existing code.